test/extended/router/h2spec.go

```diff
@@ -0,0 +1,286 @@
+package router
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	routeclientset "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/openshift/origin/test/extended/router/h2spec"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const h2specTimeout = 5 * time.Second
+
+type h2specFailingTest struct {
+	TestCase   *h2spec.JUnitTestCase
+	TestNumber int
+}
+
+type h2specRouteTypeTest struct {
+	routeType     string
```

Comment:
instead of `string`, how about [TLSTerminationType ](https://github.com/openshift/api/blob/master/route/v1/types.go#L236)?

Reply:
Addressed in https://github.com/openshift/origin/pull/24674/commits/798acd38e4abc3439c5e905a8a090afd838ec3b5